app/presenters/tree_builder_vms_and_templates.rb

```diff
@@ -51,12 +51,14 @@ def reparent_hidden_folders(tree)
   end
 
   def prune_rbac(tree)
-    vms = extract_vms(tree).uniq
-
-    allowed_vm_ids = Set.new(Rbac.filtered(vms).map(&:id)) # results_format == :ids
+    # results_format == :ids
+    allowed_vm_ids = if User.current_user.settings[:display_vms]
```

Comment:
@kbrock I think yours are appliance settings and the code here is per-user settings.